catalog/spatial/registry/registry-api-impl/src/test/java/org/codice/ddf/registry/api/impl/TestRegistryStorePublisher.java

```diff
@@ -0,0 +1,244 @@
+
+    private RegistryStorePublisher registryStorePublisher;
+
+    private RegistryPublicationService mockRegPubService;
+
+    private FederationAdminService mockFedAdminService;
+
+    private ServiceReference serviceReference;
+
+    private RegistryStore mockRegistryStore;
+
+    private Optional<Metacard> optMetacard;
+
+    private ServiceEvent mockServiceEvent;
+
+    private BundleContext bundleContext;
+
+    private Metacard mockIdentity;
+
+    private Attribute registryId;
+
+    private Event event;
+
+    @Before
+    public void setup() {
+        registryId = new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID, "registryId");
+
+        mockRegPubService = mock(RegistryPublicationService.class);
+        mockFedAdminService = mock(FederationAdminService.class);
+        serviceReference = mock(ServiceReference.class);
+        mockRegistryStore = mock(RegistryStore.class);
+        mockServiceEvent = mock(ServiceEvent.class);
+        bundleContext = mock(BundleContext.class);
+        mockIdentity = mock(Metacard.class);
+
+        optMetacard = Optional.of(mockIdentity);
+
+        registryStorePublisher = spy(new RegistryStorePublisher() {
+            @Override
+            BundleContext getBundleContext() {
+                return bundleContext;
+            }
+        });
+
+        registryStorePublisher.setFederationAdminService(mockFedAdminService);
+        registryStorePublisher.setRegistryPublicationService(mockRegPubService);
+
+        Dictionary<String, Object> eventProperties = new Hashtable<>();
+        eventProperties.put(EventConstants.EVENT, mockServiceEvent);
+        eventProperties.put(Constants.SERVICE_PID, "servicePid");
+        event = new Event("org/osgi/framework/ServiceEvent/MODIFIED", eventProperties);
+    }
+
+    @Test
+    public void testBindRegistryStoreNoContext() {
+        registryStorePublisher = spy(new RegistryStorePublisher());
+        registryStorePublisher.bindRegistryStore(serviceReference);
+
+        verify(registryStorePublisher, Mockito.times(0)).registryPublish(any());
+    }
+
+    @Test
+    public void testBindRegistryStoreAutoPush() {
+        when(bundleContext.getService(any())).thenReturn(mockRegistryStore);
+        when(mockRegistryStore.getConfigurationPid()).thenReturn("configPid");
+        when(mockRegistryStore.isAvailable(any())).thenReturn(true);
+        when(mockRegistryStore.isAutoPush()).thenReturn(true);
+        when(mockRegistryStore.getRegistryId()).thenReturn("");
+
+        registryStorePublisher.bindRegistryStore(serviceReference);
+
+        verify(registryStorePublisher, Mockito.times(1)).registryPublish(any());
+    }
+
+    @Test
+    public void testUnbindRegistryStoreNoContext() {
+        registryStorePublisher = spy(new RegistryStorePublisher());
+        registryStorePublisher.unbindRegistryStore(serviceReference);
+
+        verify(registryStorePublisher, Mockito.times(0)).registryUnpublish(any());
+    }
+
```

Comment:
Since unbindRegistryStore attempts to remove an entry from the private registryStoreMap it will throw an exception. I will refactor this test into the bind so that the map will have the entry from the first part of the test before calling this which should avoid this exception all together.